### PR TITLE
vidig47 - Solution to lab3

### PR DESCRIPTION
**Pull request by davidiga47**

### CODE REVIEW

**Review by bardhprenkaj — APPROVED** [review]
Nice, but your naming of classes and variables is horrible :-P
irisNN when we're treating MNIST :-)
mse as a variable that contains the cross entropy loss :-)

However, you're the first to use the loss function as a minimization objective for optuna which is what one should do to have fair comparisons and performance evaluations.
